drivers/stepper/stepper_shell.c

```diff
@@ -0,0 +1,458 @@
+		return err;
+	}
+
+#ifdef CONFIG_STEPPER_SHELL_ASYNC
+	start_polling(sh);
+#endif /* CONFIG_STEPPER_SHELL_ASYNC */
+
+	err = stepper_set_target_position(dev, position, poll_signal);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_enable_constant_velocity_mode(const struct shell *sh, size_t argc,
+						     char **argv)
+{
+	const struct device *dev;
+	int err;
+	enum stepper_direction direction = {-EINVAL};
+
+	for (int i = 0; i < ARRAY_SIZE(stepper_direction_map); i++) {
+		if (strcmp(argv[arg_idx_param], stepper_direction_map[i].name) == 0) {
+			direction = stepper_direction_map[i].direction;
+		}
+	}
+	if (direction == -EINVAL) {
+		return -EINVAL;
+	}
+	uint32_t velocity = strtoul(argv[arg_idx_value], NULL, 0);
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	err = stepper_enable_constant_velocity_mode(dev, direction, velocity);
+	if (err) {
+		shell_error(sh, "Error: %d", err);
+	}
+
+	return err;
+}
+
+static int cmd_stepper_info(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int err;
+	bool is_moving = false;
+	int32_t actual_position = 0;
+	enum micro_step_resolution micro_step_res = 0;
+
+	err = parse_device_arg(sh, argv, &dev);
+	if (err < 0) {
+		return err;
+	}
+
+	shell_print(sh, "Stepper Info:");
+	shell_print(sh, "Device: %s", dev->name);
+
+	err = stepper_get_actual_position(dev, &actual_position);
+	if (err < 0) {
+		shell_warn(sh, "Failed to get actual position: %d", err);
+	} else {
+		shell_print(sh, "Actual Position: %d", actual_position);
+	}
+
+	err = stepper_get_micro_step_res(dev, &micro_step_res);
+	if (err < 0) {
+		shell_warn(sh, "Failed to get micro-step resolution: %d", err);
+	} else {
+		shell_print(sh, "Micro-step Resolution: %d", micro_step_res);
+	}
+
+	err = stepper_is_moving(dev, &is_moving);
+	if (err < 0) {
+		shell_warn(sh, "Failed to check if the motor is moving: %d", err);
+	} else {
+		shell_print(sh, "Is Moving: %s", is_moving ? "Yes" : "No");
+	}
+
+	return 0;
+}
+
+#ifdef CONFIG_STEPPER_SHELL_ASYNC
+
+void stepper_poll_thread(void *p1, void *p2, void *p3)
+{
+	ARG_UNUSED(p2);
+	ARG_UNUSED(p3);
+	const struct shell *sh = p1;
+
+	while (1) {
+		k_poll(&stepper_poll_event, 1, K_FOREVER);
+
+		if (stepper_poll_event.signal->result == STEPPER_SIGNAL_STEPS_COMPLETED) {
+			shell_print(sh, "Stepper: All steps completed");
+			k_poll_signal_reset(&stepper_signal);
+		}
+	}
+}
+
+static int start_polling(const struct shell *sh)
+{
+	if (!poll_thread_started) {
+		k_tid_t tid;
+
+		k_poll_signal_init(&stepper_signal);
+		tid = k_thread_create(&poll_thread, poll_thread_stack,
+				      CONFIG_STEPPER_SHELL_THREAD_STACK_SIZE, stepper_poll_thread,
+				      (void *)sh, NULL, NULL, CONFIG_STEPPER_SHELL_THREAD_PRIORITY,
+				      0, K_NO_WAIT);
+		if (!tid) {
+			shell_error(sh, "Cannot start poll thread");
+			return -ENOEXEC;
+		}
+
+		k_thread_name_set(tid, "stepper_poll_thread");
```

Comment:
bit silly to call a thread "thread", you see this on the list of threads, imagine if all of them had "thread" in the name... just `stepper_poll` or even better `stepper_shell`?